toc.md

```diff
@@ -428,77 +429,234 @@ An additional backlog of 936 hours of image processing jobs everyday, along with
+
+A 1:2 ratio will still generate a backlog of jobs, which is why we also need to scale horizontally.
 
 ##### Scaling horizontally
-We can instead buy more servers with less cores on each. We can buy 10 servers, each with four cores for a total of 40 cores and that will get the job done. This is way more realistic, as it would be akin to buying more worker dynos on Heroku for example.
+
+Scaling horizontally means increasing the amount of available resources by getting multiple machines. This gives us two options:
+- keeping our original configuration and get multiple identical machines. In our case, we would need 10 machines since we have a 1:10 (enqueuing:processing) ratio.
+- scale vertically first by getting a better machine, such as the one described in the previous section (20 cores, 64GB or RAM) and get multiple identical machins. In our case, we would need 2 machines since we have a 1:2 (enqueuing:processing) ratio.
+
+Whichever option is chosen, they will both even out the enqueuing:processing ratio, which was our goal from the beginning.
+By choosing any of these two options we need to consider the fact that, for this to be possible, we need to have BJP that is scalable. In the next section, we will look at how this is done for Workerholic.
 
 #### Workerholic: a scalable BJP
-Now we know that we want to scale horizontally. The question is how?
+
+Workerholic is scalable because we use Redis as a central data store for the jobs. Workerholic will still need access to the source code of the main application, but won't be tied to a specific instance of the main application. Meaning, there can be a cluster of web servers or a single web server and Workerholic will still work the same. It is only concerned with what is enqueued in Redis, not how many main application instances are running. Workerholic is also scalable because its workers only don't have a state except for the queue they're polling from, which is centralized with Redis and dynamically updated every second on the Wokerholic's side.
 
 ![scalibility_workerholic](/images/scalibility_workerholic.png)
 
-Well, that's easy. Workerholic is already scalable because we use Redis as a central data store for your jobs. Workerholic will still need access to the source code of your application, but Workerholic is not tied to a specific instance of your application, so you can have distributed web servers or a single web server and Workerholic will still work just fine. Workerholic can do this because its workers only care about the queue they're polling from, which is centralized with Redis.
+Workerholic is reliable, it uses concurrency, parallelism to be efficient and it is a scalable BJP. How does it compare to a gold standard like Sidekiq? What other optimizations can we introduce?
```

Comment:
Workerholic is reliable, scalable and employs concurrency and parallelism to be efficient. How does it compare to a gold standard like Sidekiq? What other optimizations can we introduce?